src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/service/SessionSummaryService.kt

```diff
@@ -0,0 +1,81 @@
+package uk.gov.justice.digital.hmpps.educationandworkplanapi.app.service
+
+import mu.KotlinLogging
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.domain.learningandworkprogress.induction.InductionScheduleStatus
+import uk.gov.justice.digital.hmpps.domain.learningandworkprogress.induction.service.InductionScheduleService
+import uk.gov.justice.digital.hmpps.domain.learningandworkprogress.review.ReviewScheduleStatus
+import uk.gov.justice.digital.hmpps.domain.learningandworkprogress.review.service.ReviewScheduleService
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.app.client.prisonersearch.Prisoner
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.resource.model.SessionSummaryResponse
+import java.time.LocalDate
+
+private val log = KotlinLogging.logger {}
+
+@Service
+class SessionSummaryService(
+  private val prisonerSearchApiService: PrisonerSearchApiService,
+  private val reviewScheduleService: ReviewScheduleService,
+  private val inductionScheduleService: InductionScheduleService,
+) {
+
+  fun getSessionSummaries(prisonId: String): SessionSummaryResponse {
+    val today = LocalDate.now()
+    val prisoners = prisonerSearchApiService.getAllPrisonersInPrison(prisonId)
+
+    val dueReviews = mutableListOf<Prisoner>()
+    val overdueReviews = mutableListOf<Prisoner>()
+    val exemptReviews = mutableListOf<Prisoner>()
+    val dueInductions = mutableListOf<Prisoner>()
+    val overdueInductions = mutableListOf<Prisoner>()
+    val exemptInductions = mutableListOf<Prisoner>()
+
+    prisoners.forEach { prisoner ->
+      val inductionSchedule =
+        safelyGet { inductionScheduleService.getInductionScheduleForPrisoner(prisoner.prisonerNumber) }
```

Comment:
Not that I mind much 😁 , but (in the way we are calling it), is this largely the same as kotlin's`runCatching` ?
(`safelyGet` does do what it says on the tin 👍 )